packages/next/next-server/server/incremental-cache.ts

```diff
@@ -85,13 +85,15 @@ export class IncrementalCache {
     return path.join(this.incrementalOptions.pagesDir!, `${pathname}.${ext}`)
   }
 
-  private calculateRevalidate(pathname: string): number | false {
+  private calculateRevalidate(
+    pathname: string,
+    fromTime: number
+  ): number | false {
     pathname = toRoute(pathname)
 
     // in development we don't have a prerender-manifest
     // and default to always revalidating to allow easier debugging
-    const curTime = new Date().getTime()
-    if (this.incrementalOptions.dev) return curTime - 1000
+    if (this.incrementalOptions.dev) return new Date().getTime() - 1000
```

Comment:
Retain the existing behaviour for development where the current time minus 1000 is used to ensure we always regenerate pages.